File: backend/app/schemas/cover_letter.py
from pydantic import BaseModel, Field
from datetime import datetime
from typing import Literal, Optional

from app.schemas.profile import LinkType

Tone = Literal["professional", "warm", "concise"]
Status = Literal["draft", "final"]


class CoverLetterGenerate(BaseModel):
    tone: Tone = "professional"
    emphasis: Optional[str] = None
    recipient_name: Optional[str] = None
    selected_links: list[LinkType] = Field(default_factory=list)


class CoverLetterUpdate(BaseModel):
    content: str


class SupportingPoint(BaseModel):
    claim: str
    resume_evidence: str


class CoverLetterSummary(BaseModel):
    id: int
    application_id: int
    tone: Tone
    status: Status
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class CoverLetterOut(CoverLetterSummary):
    content: str
    supporting_points: list[SupportingPoint] = Field(default_factory=list)
    warnings: list[str] = Field(default_factory=list)
    resume_id: int
    emphasis: Optional[str] = None
    recipient_name: Optional[str] = None
    model: Optional[str] = None
    prompt_version: Optional[str] = None
    selected_links: dict[str, str] = Field(default_factory=dict)

File: frontend/types/cover_letter.ts
export type CoverLetterTone = "professional" | "warm" | "concise";
export type CoverLetterStatus = "draft" | "final";

export interface SupportingPoint {
  claim: string;
  resume_evidence: string;
}

export interface CoverLetterSummary {
  id: number;
  application_id: number;
  tone: CoverLetterTone;
  status: CoverLetterStatus;
  created_at: string;
  updated_at?: string | null;
}

export interface CoverLetter extends CoverLetterSummary {
  content: string;
  supporting_points: SupportingPoint[];
  warnings: string[];
  resume_id: number;
  emphasis?: string | null;
  recipient_name?: string | null;
  model?: string | null;
  prompt_version?: string | null;
}
